app/src/main/java/ch/epfl/sdp/drone3d/map/gps/LocationComponentManager.kt

```diff
@@ -65,7 +65,7 @@ class LocationComponentManager(private val activity: Activity, val mapboxMap: Ma
                 renderMode = RenderMode.COMPASS
             }
 
-            if (locationService.isLocationEnabled()) {
+            if (locationService.isLocationEnabled() && locationService.getCurrentLocation() != null) {
                 MapboxUtility.zoomOnCoordinate(locationService.getCurrentLocation()!!, mapboxMap)
             }
```

Comment:
if the location can be null, I would set the return type of getCurrentLocation() to nullable (Object?)

And the code can be simplified to 

```suggestion
            if (locationService.isLocationEnabled()) {
         
  locationService.getCurrentLocation()?.let{ MapboxUtility.zoomOnCoordinate(it, mapboxMap) }
            }
```